master/test/integration/rm/agent_test.go

```diff
@@ -0,0 +1,99 @@
+//go:build integration
+// +build integration
+
+package rm
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/determined-ai/determined/master/pkg/aproto"
+	"github.com/determined-ai/determined/master/pkg/ws"
+	"github.com/determined-ai/determined/master/test/testutils"
+	"github.com/determined-ai/determined/proto/pkg/apiv1"
+
+	"github.com/gorilla/websocket"
+	"github.com/pkg/errors"
+	"gotest.tools/assert"
+)
+
+func TestAgentConnection(t *testing.T) {
```

Comment:
I'll give it a try -- this started out as a unit test, drifted over to integration tests because mimicking agent behavior seemed difficult, then ended up setting up a fake agent anyway. It probably can move back to unit tests; thanks for pointing that out :)

Reply:
Oh i just meant the name. Integration tests dont have to live in test/integration, that was the old pattern.

Reply:
also integration == needs db or elastic, really, right now

Reply:
Ah, that's really great to know -- thanks!